### ISSUE TITLE
`mix trot.server` fails when running in dev environment

### ISSUE THREAD

**Issue by yudistrange** [hexedpackets/trot#33]
I have installed trot in my elixir toy application. When I run `mix trot.server` with a ping endpoint in production environment it works fine, however when I do the same in dev environment then it fails. 

The problem seems to lie with a compile time warning that in get in a separate module in my app. This feels bonkers, why would a warning fail my trot.server. Is this by design?

**Comment by yudistrange** [comment]
Here's the stack trace if that helps
```11:45:32.347 [error] pid=<0.426.0> #PID<0.533.0> running Router terminated
Server: localhost:4000 (http)
Request: GET /ping
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Trot.LiveReload.check_reload/2
        (trot) lib/trot/live_reload.ex:33: Trot.LiveReload.check_reload({:noop, [%Mix.Task.Compiler.Diagnostic{compiler_name: "Elixir", details: nil, file: "/Users/uditkumar/Workspace/Insider/ramp-up/my_app/lib/quiz_worker.ex", message: "function all_nodes/0 is unused", position: 348, severity: :warning}]}, %Plug.Conn{adapter: {Plug.Adapters.Cowboy.Conn, :...}, assigns: %{}, before_send: [], body_params: %Plug.Conn.Unfetched{aspect: :body_params}, cookies: %Plug.Conn.Unfetched{aspect: :cookies}, halted: false, host: "localhost", method: "GET", owner: #PID<0.533.0>, params: %Plug.Conn.Unfetched{aspect: :params}, path_info: ["ping"], path_params: %{}, peer: {{127, 0, 0, 1}, 50399}, port: 4000, private: %{}, query_params: %Plug.Conn.Unfetched{aspect: :query_params}, query_string: "", remote_ip: {127, 0, 0, 1}, req_cookies: %Plug.Conn.Unfetched{aspect: :cookies}, req_headers: [{"host", "localhost:4000"}, {"user-agent", "curl/7.54.0"}, {"accept", "*/*"}], request_path: "/ping", resp_body: nil, resp_cookies: %{}, resp_headers: [{"cache-control", "max-age=0, private, must-revalidate"}], scheme: :http, script_name: [], secret_key_base: nil, state: :unset, status: nil})
        (my_app) lib/router.ex:1: Router.plug_builder_call/2
        (plug) lib/plug/adapters/cowboy/handler.ex:16: Plug.Adapters.Cowboy.Handler.upgrade/4
        (cowboy) /Users/uditkumar/Workspace/Elixir/ramp-up/my_app/deps/cowboy/src/cowboy_protocol.erl:442: :cowboy_protocol.execute/4```

**Comment by hexedpackets** [comment]
This was caused by some changes to how mix tasks are handled with 1.6. I thought I had already handled it,but missed the case of getting compilation warnings.

Its fixed now in the newly published v0.6.1.
